htdocs/core/tpl/extrafields_list_search_sql.tpl.php

```diff
@@ -64,9 +64,14 @@
 			}
 			if (is_array($crit)) {
 				$crit = implode(' ', $crit); // natural_search() expects a string
-			} elseif ($typ === 'select' and is_string($crit) and strpos($crit, ',') === false) {
-				$critSelect = "'".implode("','", array_map(array($db, 'escape'), explode(',', $crit)))."'";
-				$sql .= " AND (".$extrafieldsobjectprefix.$tmpkey." IN (".$db->sanitize($critSelect, 1).") )";
+			} elseif ($typ === 'select' and is_string($crit)) {
```

Comment:
Can you provide an example of value for 
$crit
not working for old code and that need this new code ?

Reply:
The part of the condition that says `and strpos($crit, ',') === false`, so it doesn't run the block if $crit contains a comma, but the block itself expects $crit to contain commas (`explode(',', $crit)`).

So if the user selects "empty value" and "3" for instance, `$crit` will have the value `-2,3`. By removing the comma exclusion condition, it allows us to show the elements where the extrafield's value is either empty or 3.

Reply:
So I think what we anted to do was 
`and strpos($crit, ',') !== false`
instead of 
`and strpos($crit, ',') === false`

Reply:
Yes, I think this was the intention, but if we add this condition, it means we need more code to handle the specific case when the user selects only one value in the filter. I think the code in the PR handles both cases (unique value or multiple values) fine.